src/helpers.rs

```diff
@@ -456,29 +456,64 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         }
     }
 
+    /// Dispatches to appropriate implementations for reading an OsString from Memory,
+    /// depending on the interpretation target.
+    /// FIXME: Use `Cow` to avoid copies
+    fn read_os_str_from_target_str(&self, scalar: Scalar<Tag>) -> InterpResult<'tcx, OsString> {
+        let target_os = self.eval_context_ref().tcx.sess.target.target.target_os.as_str();
+        match target_os {
+            "linux" | "macos" => self.read_os_str_from_c_str(scalar).map(|x| x.to_os_string()),
+            "windows" => self.read_os_str_from_wide_str(scalar),
+            unsupported => throw_unsup_format!("OsString support for target OS `{}` not yet available", unsupported),
+        }
+    }
+
     /// Helper function to read an OsString from a null-terminated sequence of bytes, which is what
     /// the Unix APIs usually handle.
     fn read_os_str_from_c_str<'a>(&'a self, scalar: Scalar<Tag>) -> InterpResult<'tcx, &'a OsStr>
     where
         'tcx: 'a,
         'mir: 'a,
     {
-        #[cfg(target_os = "unix")]
+        #[cfg(unix)]
```

Comment:
Something is weird with the diff view... this change already landed in master. could you rebase, maybe that fixes it?

Reply:
Thank you for mentioning the issue.. I made another force-push, and now the diff view looks okay to me